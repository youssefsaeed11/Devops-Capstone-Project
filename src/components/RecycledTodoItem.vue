<template>
  <base-todo-item
    class="RecycledTodoItem"
    :item="item"
  >
    <template #actions>
      <div class="buttons">
        <button
          class="button is-small is-inverted is-primary"
          title="Restore from Trash"
          data-testid="restoreTrigger"
          @click="$emit('restore', item)"
        >
          <span class="icon">
            <span class="fa fa-history fa-lg" />
          </span>
        </button>
        <button
          class="button is-small is-inverted is-danger"
          title="Remove from Recycle Bin"
          data-testid="removeTrigger"
          @click="$emit('remove', item)"
        >
          <span class="icon">
            <span class="fa fa-trash fa-lg" />
          </span>
        </button>
      </div>
    </template>
  </base-todo-item>
</template>

<script>
import BaseTodoItem from '@/components/BaseTodoItem'

/**
 * @module RecycledTodoItem
 */
export default {
  name: 'RecycledTodoItem',
  components: { BaseTodoItem },
  props: {
    item: {
      type: Object,
      required: true
    }
  }
}
</script>

<style lang="scss" scoped>
.RecycledTodoItem .buttons {
  flex-wrap: nowrap;
}
</style>
