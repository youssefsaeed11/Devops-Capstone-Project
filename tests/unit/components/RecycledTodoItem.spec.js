import RecycledTodoItem from '@/components/RecycledTodoItem'
import { notDeletedItem } from '../data/ToDoFactory'
import { mount } from '@vue/test-utils'

/**
 * @param overrides
 * @return {{wrapper: Wrapper<Vue>, config: any}}
 */
const createWrapper = (overrides) => {
  let config = Object.assign(
    {
      propsData: {
        item: notDeletedItem()
      }
    },
    overrides
  )

  const wrapper = mount(RecycledTodoItem, config)

  return {
    config,
    wrapper
  }
}

describe('RecycledTodoItem', () => {
  it('renders the RecycledTodoItem', async () => {
    const { wrapper } = createWrapper()
    expect(wrapper).toMatchSnapshot()
  })

  it('Fires a restore event', async () => {
    const { wrapper } = createWrapper()
    wrapper.find(testid('restoreTrigger')).trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.emitted('restore')).toBeTruthy()
  })

  it('fires a remove event', async () => {
    const { wrapper } = createWrapper()
    wrapper.find(testid('removeTrigger')).trigger('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.emitted('remove')).toBeTruthy()
  })
})
