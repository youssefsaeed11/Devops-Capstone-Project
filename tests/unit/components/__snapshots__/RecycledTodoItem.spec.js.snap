// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecycledTodoItem renders the RecycledTodoItem 1`] = `
<div class="BaseTodoItem RecycledTodoItem">
  <div class="BaseTodoItem__checkbox"><input type="checkbox" disabled="disabled"></div>
  <div class="BaseTodoItem__content">
    notDeleted
  </div>
  <div class="BaseTodoItem__actions">
    <div class="buttons"><button title="Restore from Trash" data-testid="restoreTrigger" class="button is-small is-inverted is-primary"><span class="icon"><span class="fa fa-history fa-lg"></span></span></button> <button title="Remove from Recycle Bin" data-testid="removeTrigger" class="button is-small is-inverted is-danger"><span class="icon"><span class="fa fa-trash fa-lg"></span></span></button></div>
  </div>
</div>
`;
